dev/io.openliberty.io.smallrye.reactive.mutiny/src/io/smallrye/mutiny/infrastructure/Infrastructure.java

```diff
@@ -0,0 +1,612 @@
+/*******************************************************************************
+ * Copyright (c) contributors to https://github.com/smallrye/smallrye-mutiny
+ * Copyright (c) 2025 IBM Corporation and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Original version: https://github.com/smallrye/smallrye-mutiny/blob/2.7.0/implementation/src/main/java/io/smallrye/mutiny/infrastructure/Infrastructure.java
+ *
+ * Modified for Liberty to support SecurityManager
+ *******************************************************************************/
+
+package io.smallrye.mutiny.infrastructure;
+
+import static io.smallrye.mutiny.helpers.ParameterValidation.nonNull;
+import static io.smallrye.mutiny.helpers.ParameterValidation.positive;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Flow;
+import java.util.concurrent.Flow.Subscriber;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.BinaryOperator;
+import java.util.function.BooleanSupplier;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.LongConsumer;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.function.UnaryOperator;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.groups.MultiOverflowStrategy;
+import io.smallrye.mutiny.subscription.UniSubscriber;
+import io.smallrye.mutiny.tuples.Functions;
+
+public class Infrastructure {
+
+    private static final String DISABLE_CALLBACK_DECORATORS_PROP_NAME = "mutiny.disableCallBackDecorators";
+    private static final boolean DISABLE_CALLBACK_DECORATORS = Boolean.getBoolean(DISABLE_CALLBACK_DECORATORS_PROP_NAME);
```

Comment:
Just noticed this line does `Boolean.getBoolean` which will also read a system property - we need to either use `doPriv` here or keep the old workaround.

Reply:
Fixed.